h2o-algos/src/main/java/hex/tree/DHistogram.java

```diff
@@ -417,7 +417,7 @@ public void updateHisto(double[] ws, double[] cs, double[] ys, int [] rows, int
       double y = ys[k];
       assert (!Double.isNaN(y));
       double wy = weight * y;
-      double wyy = wy * y;
+      double wyy = wy * wy; // need to square (weight*y), not just y
```

Comment:
There is couple of places in this code where WYY is interpreted as "wy * y". We need to make it consistent. Either by renaming the variable or fixing other place where the same name has a different meaning.

Reply:
Yeah, this feels wrong with the weight.  However, this JIRA is the wrong place to fix it.  I am going to revert this change and write a JIRA to remind us to verify this later.  Thanks.